numpy/array_api/__init__.py

```diff
@@ -0,0 +1,181 @@
+    array.
+
+  - Indexing: Only a subset of indices supported by NumPy are required by the
+    spec. The Array object restricts indexing to only allow those types of
+    indices that are required by the spec. See the docstring of the
+    numpy.array_api.Array._validate_indices helper function for more
+    information.
+
+  - Type promotion: Some type promotion rules are different in the spec. In
+    particular, the spec does not have any value-based casting. The
+    Array._promote_scalar method promotes Python scalars to arrays,
+    disallowing cross-type promotions like int -> float64 that are not allowed
+    in the spec. Array._normalize_two_args works around some type promotion
+    quirks in NumPy, particularly, value-based casting that occurs when one
+    argument of an operation is a 0-d array.
+
+- All functions include type annotations, corresponding to those given in the
+  spec (see _typing.py for definitions of some custom types). These do not
+  currently fully pass mypy due to some limitations in mypy.
+
+- Dtype objects are just the NumPy dtype objects, e.g., float64 =
+  np.dtype('float64'). The spec does not require any behavior on these dtype
+  objects other than that they be accessible by name and be comparable by
+  equality, but it was considered too much extra complexity to create custom
+  objects to represent dtypes.
+
+- The wrapper functions in this module do not do any type checking for things
+  that would be impossible without leaving the array_api namespace. For
+  example, since the array API dtype objects are just the NumPy dtype objects,
+  one could pass in a non-spec NumPy dtype into a function.
+
+- All places where the implementations in this submodule are known to deviate
+  from their corresponding functions in NumPy are marked with "# Note"
+  comments. Reviewers should make note of these comments.
+
+Still TODO in this module are:
+
+- Device support and DLPack support are not yet implemented. These require
+  support in NumPy itself first.
+
+- The a non-default value for the `copy` keyword argument is not yet
+  implemented on asarray. This requires support in numpy.asarray() first.
+
+- Some functions are not yet fully tested in the array API test suite, and may
+  require updates that are not yet known until the tests are written.
+
+"""
+
+import sys
+# numpy.array_api is 3.8+ because it makes extensive use of positional-only
+# arguments.
+if sys.version_info < (3, 8):
+    raise ImportError("The numpy.array_api submodule requires Python 3.8 or greater.")
+
+import warnings
+warnings.warn("The numpy.array_api submodule is still experimental. See NEP 47.",
+              stacklevel=2)
+
+__all__ = []
+
+from ._constants import e, inf, nan, pi
+
+__all__ += ['e', 'inf', 'nan', 'pi']
+
+from ._creation_functions import asarray, arange, empty, empty_like, eye, from_dlpack, full, full_like, linspace, meshgrid, ones, ones_like, zeros, zeros_like
```

Comment:
You should shorten these lines by putting them in `(...)` to break them across line. Something like
```
from ._creation_functions import (
    asarray, arange, empty, empty_like, eye, from_dlpack, full,
    full_like, linspace, meshgrid, ones, ones_like, zeros, zeros_like
)